### PR TITLE
Switch nodeID to use tmhash.Size, add test names for net addr tests

### PR DESCRIPTION
**Pull request by ValarDragon**
Both of these came up when locally trying to change tmhash size. The lack of test names made debugging fairly confusing with all the deadbeefs. This was split into a separate PR from the rest of the hash size updates was to make that PR easier to review.

* [x] Updated all relevant documentation in docs - n/a no functionality change
* [x] Updated all code comments where relevant
* [x] Wrote tests

### CODE REVIEW

**Comment by ebuchman** [comment]
I thought we might change tmhash to be 32 bytes but still keep addresses at 20?

**Comment by ValarDragon** [comment]
@ebuchman Yeah I think thats what we're going to do. I started that change locally and just wanted to make this PR to make this easier to debug in the future. 

I think that the number of bytes in the user address should actually be 22 bytes as I noted in the issue. I'm currently discsussing with Jae.

File: p2p/netaddress_test.go [hunk 1]
```diff
@@ -22,49 +22,54 @@ func TestNewNetAddress(t *testing.T) {
...
+		{"correct nodeId", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", true},
+
+		{"too short nodeId w/tcp", "tcp://deadbeef@127.0.0.1:8080", "", false},
+		{"too short notHex nodeId w/tcp", "tcp://this-isnot-hex@127.0.0.1:8080", "", false},
+		{"notHex nodeId w/tcp", "tcp://xxxxbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "", false},
+		{"correct nodeId w/tcp", "tcp://deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", true},
+
+		{"no node id when expected", "tcp://@127.0.0.1:8080", "", false},
+		{"no node id or IP", "tcp://@", "", false},
+    {"tcp://:26656", "", false},
```
Comment (ebuchman) [inline p2p/netaddress_test.go]: Ach sorry, I need to run, can you fix this one? It's from another PR I merged in.

File: p2p/netaddress_test.go [hunk 2]
```diff
@@ -22,49 +22,52 @@ func TestNewNetAddress(t *testing.T) {
...
+		{"correct nodeId", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", true},
+
+		{"too short nodeId w/tcp", "tcp://deadbeef@127.0.0.1:8080", "", false},
+		{"too short notHex nodeId w/tcp", "tcp://this-isnot-hex@127.0.0.1:8080", "", false},
+		{"notHex nodeId w/tcp", "tcp://xxxxbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "", false},
+		{"correct nodeId w/tcp", "tcp://deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef@127.0.0.1:8080", true},
+
+		{"no node id when expected", "tcp://@127.0.0.1:8080", "", false},
+		{"no node id or IP", "tcp://@", "", false},
+		{"tcp no host, w/ port", "tcp://:26656", "", false},
```
Comment (phymbert) [inline p2p/netaddress_test.go]: Thanks @ValarDragon for the merge, it is clearer with named test cases. Maybe we should look forward to have Error() assertion equality.

**Pull request merged**